app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,634 @@
+                const multiselectSelectedItems = {...currentSelected};
+                if (currentSelected[typedItem.id]) {
+                    delete multiselectSelectedItems[typedItem.id];
+                } else {
+                    multiselectSelectedItems[typedItem.id] = typedItem;
+                }
+
+                setMultiselectSelected(multiselectSelectedItems);
+                break;
+            }
+            default: {
+                const currentSelected = multiselectSelected as Dictionary<DialogOption>;
+                const typedItem = item as DialogOption;
+                const multiselectSelectedItems = {...currentSelected};
+                if (currentSelected[typedItem.value]) {
+                    delete multiselectSelectedItems[typedItem.value];
+                } else {
+                    multiselectSelectedItems[typedItem.value] = typedItem;
+                }
+                setMultiselectSelected(multiselectSelectedItems);
+            }
+        }
+    }, [integrationData, multiselectSelected, isMultiselect, dataSource, close, handleSelect]);
+
+    const handleRemoveOption = (item: UserProfile | Channel | DialogOption) => {
+        switch (dataSource) {
+            case ViewConstants.DATA_SOURCE_USERS: {
+                const currentSelected = multiselectSelected as Dictionary<UserProfile>;
+                const typedItem = item as UserProfile;
+                const multiselectSelectedItems = {...currentSelected};
+                delete multiselectSelectedItems[typedItem.id];
+                setMultiselectSelected(multiselectSelectedItems);
+                return;
+            }
+            case ViewConstants.DATA_SOURCE_CHANNELS: {
+                const currentSelected = multiselectSelected as Dictionary<Channel>;
+                const typedItem = item as Channel;
+                const multiselectSelectedItems = {...currentSelected};
+                delete multiselectSelectedItems[typedItem.id];
+                setMultiselectSelected(multiselectSelectedItems);
+                return;
+            }
+            default: {
+                const currentSelected = multiselectSelected as Dictionary<DialogOption>;
+                const typedItem = item as DialogOption;
+                const multiselectSelectedItems = {...currentSelected};
+                delete multiselectSelectedItems[typedItem.value];
+                setMultiselectSelected(multiselectSelectedItems);
+            }
+        }
+    };
+
+    const getChannels = useCallback(debounce(async () => {
+        if (next && !loading && !term) {
+            setCurrentPage(currentPage + 1);
+            const {channels: channelData} = await fetchChannels(serverUrl, currentTeamId, currentPage);
+
+            if (channelData && channelData.length > 0) {
+                loadedChannels([...integrationData as Channel[], ...channelData]);
+            }
+        }
+    }, 100), [integrationData]);
```

Comment:
Seems weird that this only change when `integrationData` changes. What about when `next`, `loading`, `term`, `currentPage` changes?